share/spack/spackos-bootstrap/stage4/spack.in.yaml

```diff
@@ -0,0 +1,75 @@
+# This is a Spack Environment file.
+#
+# It describes a set of packages to be installed, along with
+# configuration settings.
+spack:
+  include:
+    - compilers.yaml
+  # add package specs to the `specs` list
+  specs:
+  # glibc from stage3, must match the gcc built there or we get two mixed
+  # together, use hash
+  - glibc @GLIBC_HASH@
```

Comment:
That's a good idea, it seems like it should work, I'll check.  Would be really cool to not need to treat these like templates.

Reply:
Sadly this does not work.  Setting buildable:false forces spack to treat it as an external, but it isn't.  If I made it one, then it would work but then we have the issue that it's an external.

Reply:
On an upside, after reworking things to use requirements rather than preferences, it actually doesn't seem necessary to pin the hash anymore.  May end up in the same place anyway. 🤷